parcels/_datasets/structured/generated.py

```diff
@@ -18,3 +20,267 @@ def simple_UV_dataset(dims=(360, 2, 30, 4), maxdepth=1, mesh_type="spherical"):
             "lon": (["XG"], np.linspace(-max_lon, max_lon, dims[3]), {"axis": "X", "c_grid_axis_shift": -0.5}),
         },
     )
+
+
+def radial_rotation_dataset(xdim=200, ydim=200):  # Define 2D flat, square fieldset for testing purposes.
+    lon = np.linspace(0, 60, xdim, dtype=np.float32)
+    lat = np.linspace(0, 60, ydim, dtype=np.float32)
+
+    x0 = 30.0  # Define the origin to be the centre of the Field.
+    y0 = 30.0
+
+    U = np.zeros((2, 1, ydim, xdim), dtype=np.float32)
+    V = np.zeros((2, 1, ydim, xdim), dtype=np.float32)
+
+    omega = 2 * np.pi / 86400.0  # Define the rotational period as 1 day.
+
+    for i in range(lon.size):
+        for j in range(lat.size):
+            r = np.sqrt((lon[i] - x0) ** 2 + (lat[j] - y0) ** 2)
+            assert r >= 0.0
+            assert r <= np.sqrt(x0**2 + y0**2)
+
+            theta = np.arctan2((lat[j] - y0), (lon[i] - x0))
+            assert abs(theta) <= np.pi
+
+            U[:, :, j, i] = r * np.sin(theta) * omega
+            V[:, :, j, i] = -r * np.cos(theta) * omega
+
+    return xr.Dataset(
+        {"U": (["time", "depth", "YG", "XG"], U), "V": (["time", "depth", "YG", "XG"], V)},
+        coords={
+            "time": (["time"], [np.timedelta64(0, "s"), np.timedelta64(10, "D")], {"axis": "T"}),
+            "depth": (["depth"], np.array([0.0]), {"axis": "Z"}),
+            "YC": (["YC"], np.arange(ydim) + 0.5, {"axis": "Y"}),
+            "YG": (["YG"], np.arange(ydim), {"axis": "Y", "c_grid_axis_shift": -0.5}),
+            "XC": (["XC"], np.arange(xdim) + 0.5, {"axis": "X"}),
+            "XG": (["XG"], np.arange(xdim), {"axis": "X", "c_grid_axis_shift": -0.5}),
+            "lat": (["YG"], lat, {"axis": "Y", "c_grid_axis_shift": 0.5}),
+            "lon": (["XG"], lon, {"axis": "X", "c_grid_axis_shift": -0.5}),
+        },
+    )
+
+
+def moving_eddy_dataset(xdim=2, ydim=2):  # TODO check if this also works with xdim=1, ydim=1
```

Comment:
Not sure what this TODO comment means - would a dim = 1 dataset be useful?

Reply:
Yes, because the field has no spatial variability, it would also make sense on xdim=1. But v4-dev currently doesn't support it yet. #2122 does support it (I think), so that's why I left the TODO here